New "Upgrade to v0.23 doc", changed existing Upgrade doc to be patch-specific (#200)
astronomer/docs: enterprise/v0.16/06_manage-astronomer/06_upgrade-to-0-23.md

---
title: "Upgrade to Astronomer Enterprise v0.23"
navTitle: "Upgrade to v0.23"
description: "How to upgrade the Astronomer Enterprise Platform."
---

## Overview

> As of Astronomer v0.16, Astronomer releases will be made generally available to Enterprise customers on a quarterly basis as part of a long-term support (LTS) release model.

Reviewer: Make sure we decided quarterly, and specify how long they are supported for. e.g. if we do quarterly and yearly support, then we will need to support 4 at a time. I think we might consider bi quarterly with yearly support.

> This guide walks through the process of upgrading your Astronomer Enterprise platform from 0.16.x to [v0.23](https://www.astronomer.io/docs/enterprise/0.23/resources/release-notes), which is the latest minor version.

Reviewer 2: ```suggestion
This guide walks through the process of upgrading your Astronomer Enterprise platform from 0.16.x to [v0.23](https://www.astronomer.io/docs/enterprise/0.23/resources/release-notes), which is the latest available LTS release.
```

A few notes before you start:

> - You must be on Astronomer Enterprise v0.16.x in order to complete this setup. If you are running a version of Astronomer that's lower than v0.16, submit a request to Astronomer Support. Our team will help you upgrade from your earlier version to v0.23.

Reviewer 2: ```suggestion
- You must be on Astronomer Enterprise v0.16.x in order to upgrade to Astronomer 0.23+. If you are running a version of Astronomer that's lower than v0.16, submit a request to [Astronomer Support](https://support.astronomer.io) and our team will help you define an alternate upgrade path.
```

> - The following setup is only for upgrading to the latest minor version. To upgrade to the latest patch version, read [Upgrade to a Patch Version of Astronomer Enterprise](https://www.astronomer.io/docs/enterprise/v0.16/manage-astronomer/upgrade-astronomer).

Reviewer 2: ```suggestion
- The guidelines below only apply to users who are upgrading to the Astronomer v0.23 series for the first time. Once you've completed the upgrade to any v0.23 version, you'll be free to upgrade to subsequent v0.23.x patch versions as they are released by our team. For instructions, read [Upgrade to a Patch Version of Astronomer Enterprise](https://www.astronomer.io/docs/enterprise/v0.16/manage-astronomer/upgrade-astronomer).
```

## Step 1: Check Version Compatibility

Ensure that the following software is updated to the appropriate version:

- **Kubernetes**: Your version must be greater than or equal to 1.14 and less than 1.19. If you need to upgrade Kubernetes, contact your cloud provider's support or your Kubernetes administrator.
- **Airflow Images**: You must be using an Astronomer Certified Airflow image, and the version of your image must be 1.10.5 or greater. In addition, your image should be in the following format:
```
astronomerinc/ap-airflow:<airflow-version>-<build-number>-<distribution>-onbuild
```
For example, all of the following images would work for this upgrade:
```sh
astronomerinc/ap-airflow:1.10.12-1-alpine3.10-onbuild
astronomerinc/ap-airflow:1.10.12-1-alpine3.10
astronomerinc/ap-airflow:1.10.5-9-buster-onbuild
astronomerinc/ap-airflow:1.10.5-9-buster
```

> > **Note:** While `-onbuild` and `<build-number>` are optional, we recommend including them for most upgrades. If you have your own build, test, and publish workflows that are layered on top of the Astronomer Airflow images, then removing `<build-number>` is appropriate.

Reviewer: ```suggestion
> **Note:** While `-onbuild` and `<build-number>` are optional, we recommend including them for most upgrades. If you have your own build, test, and publish workflows that are layered on top of the Astronomer Airflow images, then removing `<build-number>` is appropriate. The images including build number are immutable.
```

## Step 2: Check Permissions

Minor version upgrades can be initiated only by a user with System Admin permissions on Astronomer. To confirm you're an Astronomer SysAdmin, confirm that you have access to **System Admin** features in the in the Astronomer UI:

![Admin](https://assets2.astronomer.io/main/docs/enterprise_quickstart/admin_panel.png)

You also need permission to create Kubernetes resources. To confirm you have those permissions, run the following commands:

```
$ kubectl auth can-i create pods --namespace <your-astronomer-namespace>
$ kubectl auth can-i create sa --namespace <your-astronomer-namespace>
$ kubectl auth can-i create jobs --namespace <your-astronomer-namespace>
```

If all commands return `yes`, then you have the appropriate Kubernetes permissions.

## Step 3: Backup Your Database

Backup your database using your cloud provider's functionality for doing so, or make a request to your database administrator to backup based on your organization's own guidelines.

## Step 4: Check the Status of Your Kubernetes Pods

> All pods should be either `Running`, `Completed`, or in `CrashLoopBackOff`. If you have any pods that are crashing, but this is expected behavior and you want to proceed anyways, make note of which ones before before proceeding.

Reviewer 2: I'd still rephrase this to say:

> Before you proceed with the upgrade, ensure that the Kubernetes Pods in your platform namespace are healthy. To do so, run:
> 
> ```
> $ kubectl get pods -n <your-astronomer-namespace>
> ```
> 
> All pods should be in either the `Running` or `Completed` state. If any of your pods are in a `CrashLoopBackOff` state or are otherwise unhealthy, make sure that's expected behavior before you proceed.

Author: Sounds good, was not quite understanding the `CrashLoopBackOff` part of things before, but this clears it up. Updated!

```sh
$ kubectl get pods | grep upgrade-astronomer
```

Then, run the following command for each pod you find:

```sh
$ kubectl logs <your-pod-name>
```

## Step 7: Confirm That the Upgrade Was Successful

If the upgrade was successful, you should be able to:

* Log in to Astronomer at `astronomer.BASEDOMAIN`.
* See Workspaces and Airflow Deployments in the Astronomer UI.
* Access the **Settings** tab for each of your Deployments in the Astronomer UI.
* See metrics on the **Metrics** tab in the Astronomer UI.
* Successfully run `$ astro deploy` using the Astronomer CLI.
* Open the Airflow UI for each of your Deployments
* Access logs for your DAGs in the Airflow UI.

## Roll Back to Enterprise v0.16

If you encounter an issue during your upgrade that requires you to recover your original platform, you can roll back to Astronomer v0.16. To do so:

1. Apply the rollback automation script by running the following command:
```sh
kubectl apply -f https://raw.githubusercontent.com/astronomer/astronomer/master/bin/migration-scripts/lts-to-lts/0.16-to-0.23/manifests/rollback.yaml

> ```

Reviewer 2: @sjmiller609 Can you shed more insight on what this script calls for? Are Metadata Databases for all Airflow Deployments preserved?

Reviewer: it restores the platform DB and the helm state of the Astronomer helm chart

> 

Reviewer 2: ```suggestion
This restores the Platform Database and the helm state of the Astronomer helm chart.

```

Author: Took the capitalization out of "Platform Database", unless that's an official Astronomer term?

---
title: "Upgrade to Astronomer Enterprise v0.23"
navTitle: "Upgrade to v0.23"
description: "How to upgrade the Astronomer Enterprise Platform."
---

## Overview

As of Astronomer v0.16, Astronomer releases will be made generally available to Enterprise customers on a quarterly basis as part of a long-term support (LTS) release model.

> This guide walks through the process of upgrading your Astronomer Enterprise platform to v0.23, which is the latest minor version.

Reviewer 2: Link out to 0.23 release notes? This will live in `https://www.astronomer.io/docs/enterprise/0.23/resources/release-notes` (next)

> This guide walks through the process of upgrading your Astronomer Enterprise platform to v0.23, which is the latest minor version.

Reviewer 2: ```suggestion
This guide walks through the process of upgrading your Astronomer Enterprise platform from 0.16.x to v0.23, which is the latest minor version.
```

> This guide walks through the process of upgrading your Astronomer Enterprise platform to v0.23, which is the latest minor version.

Reviewer 2: I would also add a note: 

> If you are running a version of Astronomer that's lower than [v0.16 (latest)](/docs/enterprise/v0.16/resources/release-notes),  submit a request to [Astronomer Support](https://support.astronomer.io). Our team will help you determine an upgrade path to Astronomer 0.16 first before you can upgrade to Astronomer v0.23.

(Sort of weirdly worded - but something like that)

## Astronomer Platform Versioning

Astronomer platform releases follow a semantic versioning scheme. All versions are written as a 3-component number in the format of `x.y.z`. In this syntax,

- X: Major Version
- Y: Minor Version
- Z: Patch/Hotfix

For example, upgrading Astronomer from v0.16.4 to v0.16.5 would be considered upgrading to a patch version, whereas upgrading from v0.15.0 to v0.16.0 would be considered upgrading to the latest minor version.

The following setup is only for upgrading to the latest minor version. To upgrade to a patch version, read [Upgrade to a Patch Version of Astronomer Enterprise](https://www.astronomer.io/docs/enterprise/v0.16/manage-astronomer/upgrade-astronomer)

## Step 1: Check Version Compatibility

Ensure that the following software

- **Kubernetes**: Your version must be greater than or equal to 1.14 and less than 1.19. If you need to upgrade Kubernetes, contact your cloud provider's support or your Kubernetes administrator.

> - **Airflow Images**: You must be using an Astronomer Certified Airflow Image, and the version of your Image must be 1.10.5 or greater. In addition, your image should be in the following format:

Reviewer 2: I'm actually not sure "Image" needs to be capitalized always - do you know? Double check?

Author: Kubernetes does not capitalize it, so that seems like the move!

```
astronomerinc/ap-airflow:<airflow-version>-<build-number>-<distribution>-onbuild
```
For example, all of the following images would work for this upgrade:
```sh
astronomerinc/ap-airflow:1.10.12-1-alpine3.10-onbuild
astronomerinc/ap-airflow:1.10.12-1-alpine3.10
astronomerinc/ap-airflow:1.10.5-9-buster-onbuild
astronomerinc/ap-airflow:1.10.5-9-buster
```
> **Note:** While `-onbuild` and `<build-number>` are optional, we recommend including them for most upgrades. If you have your own build, test, and publish workflows that are layered on top of the Astronomer Airflow images, then removing `<build-number>` is appropriate.

> 

Reviewer 2: Spaces missing in the above section? E.g. after `format:` , before `For example`, and before "Note"

Author: Because this is a bulleted list, we need to keep the spacing tight so the notes on Airflow images appear under the Airflow images bullet.

## Step 2: Check Permissions

> Minor version upgrades can be initiated only by Astronomer Admins. To confirm you're an Astronomer Admin, confirm that you have access to **System Admin** features in the in the Astronomer UI:

Reviewer 2: ```suggestion
Minor version upgrades can be initiated only by a user with System Admin permissions on Astronomer. To confirm you're an Astronomer SysAdmin, confirm that you have access to **System Admin** features in the in the Astronomer UI:
```

![Admin](https://assets2.astronomer.io/main/docs/enterprise_quickstart/admin_panel.png)

You also need permission to create Kubernetes resources. These permissions can be confirmed by running the following commands:

```
$ kubectl auth can-i create pods --namespace <your-astronomer-namespace>
$ kubectl auth can-i create sa --namespace <your-astronomer-namespace>
$ kubectl auth can-i create jobs --namespace <your-astronomer-namespace>
```

If all commands return `yes`, then you have the appropriate Kubernetes permissions.

## Step 3: Backup Your Database

Backup your database using your cloud provider's functionality for doing so, or make a request to your database administrator to backup based on your organization's own guidelines.

> 

Reviewer 2: Probably a question for @sjmiller609 - do we want/need to give users more info here? Should we link out to our Disaster Recovery doc, which talks about this in more detail?

Should we also clarify "what database"? Assuming here we're talking about Platform DB

Reviewer: the whole database instance, i.e. backing up everything

Reviewer 2: So maye -

```suggestion
Backup your entire Astronomer database instance using your cloud provider's functionality for doing so, or make a backup request to your database administrator based on your organization's guidelines.
```

## Step 4: Check the Status of Your Kubernetes Pods

> All pods should be either `Running` or `Completed`. If you have any pods that are crashing, but this is expected behavior and you want to proceed anyways, make note of which pods are crashing before upgrading.

Reviewer 2: ```suggestion
All pods should be either `Running` or `Completed`, though some may be in `CrashLoopBackOff`. If you have any pods that are crashing, make note of which ones before upgrading.
```

A little confused on the recommendation here given the first draft of this - should users proceed? Or is it fine as long as they make note of them now?

Author: I think the intention here was to ensure that readers have a way of knowing which pods are crashing because of the upgrade, vs which ones are crashing for other reasons. I tried clarifying that they can, in fact, proceed if they have crashing pods.

Reviewer 2: Got it hm, I think the wording is mixing me up here. Made another suggestion!

## Step 5: Switch to Your Default Namespace

Switch to the default namespace in your Kubernetes context by running the following command:

```sh
$ kubectl config set-context --current --namespace=default
```

## Step 6: Upgrade Astronomer

Run the following command to begin the upgrade process:

```

> kubectl apply -f https://raw.githubusercontent.com/astronomer/astronomer/master/bin/migration-scripts/lts-to-lts/0.16-to-0.23/manifests/upgrade.yaml

Reviewer 2: ```suggestion
$ kubectl apply -f https://raw.githubusercontent.com/astronomer/astronomer/master/bin/migration-scripts/lts-to-lts/0.16-to-0.23/manifests/upgrade.yaml
```

```

While your platform is upgrading, monitor your pods to ensure that no errors occur. To do so, first find the names of your pods by running the following command:

```sh
$ kubectl get pods | grep upgrade-astronomer
```

> Then, run the following command to see each pod while you upgrade:

Reviewer 2: ```suggestion
Then, run the following command to follow the logs of any pod while you upgrade:
```

```sh
$ kubectl logs <your-pod-name>
```

## Step 7: Confirm That the Upgrade Was Successful

If the upgrade was successful, you should be able to:

* Log in to Astronomer at `astronomer.BASEDOMAIN`
* See Workspaces and Airflow Deployments in the Astronomer UI

> * Access the settings of your Airflow Deployments

Reviewer 2: What are the "settings of your Airflow Deployments"? Is that the **Settings** tab in the Astronomer UI?

Author: I think so, but @stevenjs1 can confirm.

Reviewer 2: @jwitz Maybe then rephrase to: `Access the **Settings** tab of all Airflow Deployments`

* See metrics on the **Metrics** tab in the Astronomer UI
* Successfully run `$ astro deploy` using the Astronomer CLI

> * Open a Deployment in the Airflow UI

Reviewer 2: ```suggestion
* Open the Airflow UI of any Deployment
```

* Access logs for your DAGs in the Airflow UI

## Roll Back to Enterprise v0.16

> If the upgrade has some issues and you need to recover your platform, you can roll back to v0.16. To do so:

Reviewer 2: ```suggestion
If you encounter an issue during your upgrade such that you'd like to recover the platform, you may choose to roll back to Astronomer v0.16. To do so:
```

Author: Updated this wording slightly in my latest commit.
